### PR TITLE
LPS-147808 [CP2.0] Improve how roles are disabled when user reaches role limit - error message

### PR DESCRIPTION
**Pull request by claricebrasil**
### How it was developed?
Was created a new function to handle if the user can add a new team member or not. This should happen only if the user fill the e-mail field. If the user try to add a new Team Member without filling the e-mail field, should show an error message under the email field. To show this message, we created a logic to only show the Badge if the state that says if the field email is empty or not, is true.

### CODE REVIEW

**Review by Lucas2691 — APPROVED** [review]
Good job.

**Review by thaynam — APPROVED** [review]

**Review by joanagrego — APPROVED** [review]
Good job!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 1]
```diff
@@ -181,7 +182,14 @@ const InviteTeamMembersPage = ({
...
 			}
-
+			else {
+				setAccountRolesOptions((previousAccountRoles) =>
+					previousAccountRoles.map((previousAccountRoles) => ({
+						...previousAccountRoles,
+						disabled: false,
+					}))
+				);
+			}
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:192]: Avoid else statment, use return instead.
To do so, return an empty value (just return;) at the end of the first if statment (line 90)
```suggestion
			
				setAccountRolesOptions((previousAccountRoles) =>
					previousAccountRoles.map((previousAccountRoles) => ({
						...previousAccountRoles,
						disabled: false,
					}))
				);
			
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 2]
```diff
@@ -285,83 +313,132 @@ const InviteTeamMembersPage = ({
...
+							</div>
+						</div>
+						<div className="invites-helper px-3">
+							<div className="mx-3 pt-3">
+								<h5 className="text-neutral-7">
+									{`${
+										projectHasSLAGoldPlatinum
+											? ROLE_TYPES.requestor.name
+											: ROLE_TYPES.admin.name
+									}	roles available: ${availableAdminsRoles} of ${maxRequestors}`}
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:418]: Create a variable for that

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/customer/fragments/side-menu/index.html [hunk 3]
```diff
@@ -4,22 +4,22 @@
...
 					<use xlink:href="${clayIconSVG}#angle-right-small" />
 				</svg>
 			</button>
 
-			<ul class="list-unstyled ml-3 overflow-hidden" id="customer-portal-products" />
+		<ul class="list-unstyled ml-3 overflow-hidden" id="customer-portal-products" />
 		</li>
 		<li>
-			<button class="btn btn-menu btn-sm customer-portal-side-menu-button skeleton" id="custumer-portal-members" type="button" value="team">
+		<button class="btn btn-menu btn-sm customer-portal-side-menu-button skeleton" id="custumer-portal-members" type="button" value="team_members">
```
Comment (iagochaves) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/customer/fragments/side-menu/index.html:22]: Hey @claricebrasil, we do not use side menu as a fragment anymore, please modify the SideMenu React component

**Review by iagochaves — CHANGES_REQUESTED** [review]
Cool, just a few suggestions

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js (hunk 1 above)
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:192]: Nice, thanks for the tip!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js (hunk 2 above)
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:418]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 4]
```diff
@@ -252,6 +259,25 @@ const InviteTeamMembersPage = ({
...
+
+		setshowEmptyEmailError(emptyEmailAddress);
+
+		return !emptyEmailAddress;
+	};
+
+	useEffect(() => {
+		if (availableAdminsRoles > 0) {
+			setAccountRolesOptions((previousAccountRoles) =>
+				previousAccountRoles.map((previousAccountRoles) => ({
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:273]:
```suggestion
				previousAccountRoles?.map((previousAccountRoles) => ({
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/customer/fragments/side-menu/index.html [hunk 5]
```diff
@@ -4,22 +4,22 @@
...
 					<use xlink:href="${clayIconSVG}#angle-right-small" />
 				</svg>
 			</button>
 
-			<ul class="list-unstyled ml-3 overflow-hidden" id="customer-portal-products" />
+		<ul class="list-unstyled ml-3 overflow-hidden" id="customer-portal-products" />
 		</li>
 		<li>
-			<button class="btn btn-menu btn-sm customer-portal-side-menu-button skeleton" id="custumer-portal-members" type="button" value="team">
+		<button class="btn btn-menu btn-sm customer-portal-side-menu-button skeleton" id="custumer-portal-members" type="button" value="team">
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/customer/fragments/side-menu/index.html:22]: Check this indentation, I guess the format change it wrong

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 6]
```diff
@@ -252,6 +259,25 @@ const InviteTeamMembersPage = ({
 		}
 	};
 
+	const handleAddTeamMember = () => {
+		const emptyEmailAddress = values?.invites?.some(({email}) => !email);
+
+		setshowEmptyEmailError(emptyEmailAddress);
+
+		return !emptyEmailAddress;
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:267]:
```suggestion
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 7]
```diff
@@ -252,6 +259,25 @@ const InviteTeamMembersPage = ({
 		}
 	};
 
+	const handleAddTeamMember = () => {
+		const emptyEmailAddress = values?.invites?.some(({email}) => !email);
+
+		setshowEmptyEmailError(emptyEmailAddress);
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:265]: Should be good choose a better variable name :)
A good practice it's always be a trusty value and negative it. Eg.: hasEmailsFilled, and add our negative value: !hasEmailsFilled

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 8]
```diff
@@ -285,93 +311,128 @@ const InviteTeamMembersPage = ({
...
+								))}
+							</ClayForm.Group>
+
+							{showEmptyEmailError && (
+								<Badge>
+									<span className="pl-1">
+										Please enter your email address.
+									</span>
+								</Badge>
+							)}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:357]: We need to use our approach to show error, integrate it with <Input> and formik. Maybe use a setError with setTouched.

Let's try avoid to use errors outside the Input
```suggestion
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 9]
```diff
@@ -285,93 +311,128 @@ const InviteTeamMembersPage = ({
...
+
+								{values?.invites?.length <
+									MAXIMUM_INVITES_COUNT && (
+									<Button
+										className="btn-outline-primary cp-btn-add-members py-2 rounded-xs"
+										onClick={() => {
+											setBaseButtonDisabled(false);
+
+											if (handleAddTeamMember()) {
+												push(
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:394]: Attach the value in some variable before use within if.
And should be good choose a better name for this function. What the purpose? I see that it's return a validation and also set in some state.

Function need to be scoped. A better approach it's rename this function to isEmptyFillEmails, or change it to a simple variable, and the state's update come inside this condition

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js [hunk 10]
```diff
@@ -69,12 +69,17 @@ const InviteTeamMembersPage = ({
 	const [accountRolesOptions, setAccountRolesOptions] = useState([]);
 	const [accountRoles, setAccountRoles] = useState([]);
 	const [availableAdminsRoles, setAvailableAdminsRoles] = useState(1);
+	const [showEmptyEmailError, setshowEmptyEmailError] = useState(false);
```
[the remark this answers is not in the document]
Comment (claricebrasil) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js:72]: Done!

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/customer/fragments/side-menu/index.html (hunk 5 above)
Comment (kevenleone) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/src/main/resources/site-initializer/fragments/customer/fragments/side-menu/index.html:22]: The format source is weird here, if you try "gw formatSource" and continues the same, you need to manually indent the file

Can you try @claricebrasil ?

**Comment by kevenleone** [comment]
@claricebrasil this message is appearing here: `Conflicting files
modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js`

Can you rebase with master?

**Comment by claricebrasil** [comment]
> @claricebrasil this message is appearing here: `Conflicting files modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/InviteTeamMembersForm/index.js`
> 
> Can you rebase with master?

Hey, Keven! I fixed it now, can you check is is ok? Thank you!!

**Comment by jairmedeiros** [comment]
ci:forward

**Pull request closed without merging**